File: WorldFeed/Server/src/WorldFeed.Common/DTO/TwitterListDTO.cs
namespace WorldFeed.Common.DTO
{
    using System;
    using Newtonsoft.Json;

    using WorldFeed.Common.JsonConverters;
    using WorldFeed.Common.Public.Models.Enums;
    using WorldFeed.Common.Public.Models.Interfaces.DTO;

    public class TwitterListDTO : ITwitterListDTO
    {
        [JsonProperty("id")]
        [JsonConverter(typeof(JsonPropertyConverterRepository))]
        public long Id { get; set; }

        [JsonProperty("id_str")]
        public string IdStr { get; set; }

        [JsonProperty("slug")]
        public string Slug { get; set; }

        [JsonIgnore]
        public long OwnerId => Owner?.Id ?? 0;

        [JsonIgnore]
        public string OwnerScreenName => Owner?.ScreenName;

        [JsonProperty("name")]
        public string Name { get; set; }

        [JsonProperty("full_name")]
        public string FullName { get; set; }

        [JsonProperty("user")]
        public IUserDTO Owner { get; set; }

        [JsonProperty("created_at")]
        [JsonConverter(typeof(JsonTwitterDateTimeConverter))]
        public DateTime CreatedAt { get; set; }

        [JsonProperty("uri")]
        public string Uri { get; set; }

        [JsonProperty("description")]
        public string Description { get; set; }

        [JsonProperty("following")]
        public bool Following { get; set; }

        [JsonProperty("mode")]
        [JsonConverter(typeof(JsonPropertyConverterRepository))]
        public PrivacyMode PrivacyMode { get; set; }

        [JsonProperty("member_count")]
        public int MemberCount { get; set; }

        [JsonProperty("subscriber_count")]
        public int SubscriberCount { get; set; }
    }
}

File: Sharebook/Client/src/app/core/Core/DTO/TwitterListDTO.ts
import {PrivacyMode} from '../../Public/Models/Enum/PrivacyMode';
import {ITwitterListDTO} from "../../Public/Models/Interfaces/DTO/ITwitterListDTO";
import {IUserDTO} from "../../Public/Models/Interfaces/DTO/IUserDTO";
import DateTime from "typescript-dotnet-commonjs/System/Time/DateTime";

export class TwitterListDTO implements ITwitterListDTO {
  // [JsonProperty("id")]
  // [JsonConverter(typeof(JsonPropertyConverterRepository))]
  public id: number;

  // [JsonProperty("id_str")]
  public idStr: string;

  // [JsonProperty("slug")]
  public slug: string;

  // [JsonIgnore]
  get ownerId(): number {
    return this.owner?.id ?? 0;
  }

  // [JsonIgnore]
  get ownerScreenName(): string {
    return this.owner?.screenName;
  }

  // [JsonProperty("name")]
  public name: string;

  // [JsonProperty("full_name")]
  public fullName: string;

  // [JsonProperty("user")]
  public owner: IUserDTO;

  // [JsonProperty("created_at")]
  // [JsonConverter(typeof(JsonTwitterDateTimeOffsetConverter))]
  public createdAt: DateTime; // DateTimeOffset`

  // [JsonProperty("uri")]
  public uri: string;

  // [JsonProperty("description")]
  public description: string;

  // [JsonProperty("following")]
  public following: boolean;

  // [JsonProperty("mode")]
  // [JsonConverter(typeof(JsonPropertyConverterRepository))]
  public privacyMode: PrivacyMode;

  // [JsonProperty("member_count")]
  public memberCount: number;

  // [JsonProperty("subscriber_count")]
  public subscriberCount: number;
}
